### PR TITLE
Fixups for name mangling in ARM

### PR DESCRIPTION
**Pull request by travitch**
This makes sure we have uf_ and df_ prefixes in the right spots.  It also
switches a Map to a HashMap in a context with strings for keys to speed up
parsing.

### CODE REVIEW

**Comment by travitch** [comment]
This change also avoids re-serializing formulas into bytestrings while loading the semantics (to speed up use of this code in macaw).

File: semmc-aarch32/src/SemMC/Architecture/ARM/ASL.hs [hunk 1]
```diff
@@ -412,7 +401,7 @@ unsafeSymbol nm = case WI.userSymbol nm of
   Right s -> s
 
 dropUFPrefix :: String -> String
-dropUFPrefix nm = case List.stripPrefix "uf." nm of
+dropUFPrefix nm = case List.stripPrefix "uf_" nm of
```
Comment (danmatichuk) [inline semmc-aarch32/src/SemMC/Architecture/ARM/ASL.hs:404]: Can probably just delete this.

Comment (travitch) [reply semmc-aarch32/src/SemMC/Architecture/ARM/ASL.hs:404]: Oh, good catch

Comment (travitch) [reply semmc-aarch32/src/SemMC/Architecture/ARM/ASL.hs]: Okay, I did a warning cleanup pass and got rid of that

**Pull request merged**